### PR TITLE
NEH: Icon links, latest PST, more source providers, and bug fixes

### LINKED ISSUE #285: Add badges and icon links to sidebar
With the alabaster theme I like that you can have badges added to the sidebar (and a description), e.g.

image

See https://github.com/bitprophet/alabaster/blob/master/alabaster/theme.conf for all options

### Implementation

The easiest way to do this would be to support the Icon Links functionality from the pydata theme:

- https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/header-links.html#icon-links

### PR DESCRIPTION
**Pull request by choldgraf**
This brings in the latest pydata theme, and along the way includes support for their `icon-links` buttons in our primary sidebar. We just inherit the component template and have some lightweight styling to make it look in-line with our documentation style.

- Brings in some more fixes from feedback in https://github.com/executablebooks/sphinx-book-theme/issues/686
- closes #285 
- Fixes the bug about `builder.theme_options` that @mmcky reported
- closes https://github.com/executablebooks/sphinx-book-theme/issues/694
- closes https://github.com/executablebooks/sphinx-book-theme/issues/606

Also adds support for more source button providers via the pydata theme:

- closes https://github.com/executablebooks/sphinx-book-theme/issues/418
- closes https://github.com/executablebooks/sphinx-book-theme/issues/114
- closes https://github.com/executablebooks/sphinx-book-theme/issues/564

### CODE REVIEW

**Comment by mmcky** [comment]
> Fixes the bug about builder.theme_options that @mmcky reported

thanks @choldgraf.

Once a new `rc` is available I will check against https://github.com/executablebooks/jupyter-book/pull/1842

File: docs/components/icon-links.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Icon links and badges
+
+You can add a collection of icon links and badges to your primary sidebar.
+For example, to include links to your GitHub repository or a [shields.io badge](https://shieds.io).
```
Comment (choldgraf) [inline docs/components/icon-links.md:4]:
```suggestion
For example, to include links to your GitHub repository or a [shields.io badge](https://shields.io).
```

File: docs/conf.py [hunk 2]
```diff
@@ -131,6 +131,25 @@
...
+        },
+        {
+            "name": "GitHub",
+            "url": "https://github.com/executablebooks/sphinx-book-theme",
+            "icon": "fa-brands fa-github",
+        },
+        {
+            "name": "PyPI",
+            "url": "https://pypi.org/project/sphinx-book-theme/",
+            "icon": "https://img.shields.io/pypi/dd/sphinx-book-theme",
```
Comment (choldgraf) [inline docs/conf.py:149]:
```suggestion
            "icon": "https://img.shields.io/pypi/dw/sphinx-book-theme",
```

**Comment by choldgraf** [comment]
OK this one now cleans up a bunch of different things, if we can get the tests passing, I'll merge so that we can cut a new release candidate since a lot of feedback was addressed and a couple new things were added.

**Comment by choldgraf** [comment]
cc @mmcky - making a new release candidate now so try it out! :-)

https://github.com/executablebooks/sphinx-book-theme/releases/tag/v1.0.0rc3

**Pull request merged**